test/oracles/UniswapV3Oracle.test.ts

```diff
@@ -0,0 +1,272 @@
+  }
+
+  afterEach(reverter.revert);
+
+  describe("init", () => {
+    it("should set oracle correctly", async () => {
+      expect(await oracle.uniswapV3Factory()).to.equal(await uniswapV3Factory.getAddress());
+    });
+
+    it("should not initialize twice", async () => {
+      await expect(oracle.mockInit(await uniswapV3Factory.getAddress())).to.be.revertedWith(
+        "Initializable: contract is not initializing"
+      );
+      await expect(oracle.__OracleV3Mock_init(await uniswapV3Factory.getAddress())).to.be.revertedWith(
+        "Initializable: contract is already initialized"
+      );
+    });
+  });
+
+  describe("getPrice", () => {
+    it("should correctly get price if there are same tokens in the path ", async () => {
+      let ans = await oracle.getPriceOfTokenInToken([A_TOKEN, A_TOKEN], [FeeAmount.MEDIUM], 51, PERIOD);
+
+      expect(ans[0]).to.equal(51);
+    });
+
+    it("should correctly get price if it older than observation", async () => {
+      pool = await createPools(A_TOKEN, B_TOKEN);
+
+      const timeFirst = (await time.latest()) + 2;
+      await time.increaseTo(timeFirst);
+
+      await pool.initialize(encodePriceSqrt(10 ** 6, 10 ** 18));
+
+      await time.increaseTo(timeFirst + 3);
+
+      let ans = await oracle.getPriceOfTokenInToken(A_B_PATH, [FeeAmount.MEDIUM], 5n * wei("1"), 5);
+
+      expect(ans[0]).to.equal(getPriceByTick(Number((await pool.slot0()).tick), 5n * wei("1")));
+      expect(ans[1]).to.equal((await time.latest()) - timeFirst - 1);
+    });
+
+    it("should correctly increase cardinality and overwrite observations", async () => {
+      pool = await createPools(A_TOKEN, B_TOKEN);
+
+      await pool.initialize(encodePriceSqrt(10 ** 3, 10 ** 12));
+      await pool.increaseObservationCardinalityNext(2);
+
+      await time.increaseTo((await time.latest()) + 1);
+
+      await pool.addObservation(-111);
+
+      const firstTime = await time.latest();
+      await time.increaseTo(firstTime + 3);
+
+      await pool.addObservation(250);
+
+      let ans = await oracle.getPriceOfTokenInToken(A_B_PATH, [FeeAmount.MEDIUM], 10n ** 6n, 3);
+
+      expect(ans[0]).to.equal(getPriceByTick(-111, 10n ** 6n));
+      expect(ans[1]).to.equal(3);
+    });
+
+    it("should correctly get price for complex path", async () => {
+      pool = await createPools(A_TOKEN, B_TOKEN);
+      let poolBC = await createPools(B_TOKEN, C_TOKEN);
+
+      await pool.initialize(encodePriceSqrt(2 * 10 ** 6, 10 ** 18));
+      await poolBC.initialize(encodePriceSqrt(10 ** 2, 10 ** 6));
+
+      await time.increaseTo((await time.latest()) + 2);
+
+      let ans = await oracle.getPriceOfTokenInToken(
+        [A_TOKEN, B_TOKEN, C_TOKEN],
+        [FeeAmount.MEDIUM, FeeAmount.MEDIUM],
+        2n * wei("1"),
+        PERIOD
+      );
+
+      //priceAB = 1.0001 ** tickAB * amount, priceBC = 1.0001 ** tickBC * amount,
+      //so priceAC = priceAB*priceBC = amount(1.0001**(tickAB+tickBC))
```

Comment:
```suggestion
      // so priceAC = priceAB * priceBC = amount * (1.0001 ** (tickAB + tickBC))
```
Cosmetic for all comments, please add spaces after "//"